### DISCUSSION TITLE
Sync value of different fields in a collection?

### DISCUSSION THREAD

**Discussion by hunghvu** [payloadcms/payload#900] [General]
There are fields that act multiple roles on a page. For example, let's say we have a field `pageTitle`, it can be:
1.  Article Title
2. SEO Title
3. `og:title`
4. schema.org title

Now, we can have only 1 field `pageTitle` in the admin editor, and leave usage implementation of the field to the front end. For example:

```jsx
// JSON response. This means the developer has to know the value is used in different places.
{
  pageTitle: "How to use Payload CMS?"
}

// Presentation layer, pseudo-code. 
<meta property="og:title" content={response.pageTitle}>
<title>{response.pageTitle}</title>
<ArticleTitleComponent>{response.pageTitle}</ArticleTitleComponent>
...
```

Now, what about we have different fields instead? This may better show the intention of the data presented in the response.

```jsx
// JSON response. The developer knows what to implement just from the response.
{
  pageTitle: "How to use Payload CMS?"
  ogTitle: "How to use Payload CMS?"
  articleTitle: "How to use Payload CMS?"
}

// Presentation layer, pseudo code
<meta property="og:title" content={response.ogTitle}>
<title>{response.pageTitle}</title>
<ArticleTitleComponent>{response.articleTitle}</ArticleTitleComponent>
...
```

With that said, is there a native way to sync the values of different fields in the editor, something that resembles `admin.usedAsTitle`? Manually typing the value to different fields in the editor is certainly not desirable.

**Comment by TimHal** [comment]
You can have several readonly fields (and even hide them from the admin view) if you want. Synchronization can be done with [FieldHooks](https://payloadcms.com/docs/hooks/fields). The Hook can be used to update all the readonly fields whenever the "primary" pageTitle changes.

**Reply by hunghvu** [reply]
This sounds promising and is exactly what I want. I will take a look into the hooks.
